pkg/client/kubeconfig_test.go

```diff
@@ -0,0 +1,270 @@
+package client
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"math/big"
+	"os"
+	"strings"
+	"time"
+
+	mocks "github.com/Azure/aks-tls-bootstrap-client/pkg/client/mocks"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"go.uber.org/mock/gomock"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+var _ = Describe("TLS Bootstrap kubeconfig tests", func() {
+	var (
+		mockCtrl            *gomock.Controller
+		imdsClient          *mocks.MockImdsClient
+		aadClient           *mocks.MockAadClient
+		tlsBootstrapClient  *tlsBootstrapClientImpl
+		mockReader          *mocks.MockfileReader
+		validKubeConfigPath string
+	)
+
+	Context("isKubeConfigStillValid Tests", func() {
+		BeforeEach(func() {
+			validKubeConfigPath = "dummykubeconfig"
+			_, err := os.Create(validKubeConfigPath)
+
+			if err != nil {
+				return
+			}
+
+			mockCtrl = gomock.NewController(GinkgoT())
+			imdsClient = mocks.NewMockImdsClient(mockCtrl)
+			aadClient = mocks.NewMockAadClient(mockCtrl)
+			mockReader = mocks.NewMockfileReader(mockCtrl)
+
+			tlsBootstrapClient = &tlsBootstrapClientImpl{
+				logger:     testLogger,
+				imdsClient: imdsClient,
+				aadClient:  aadClient,
+				reader:     mockReader,
+			}
+		})
+
+		AfterEach(func() {
+			err := os.Remove(validKubeConfigPath)
+			if err != nil {
+				return
+			}
+
+			mockCtrl.Finish()
+		})
+
+		When("kubeconfig is valid", func() {
+			It("should return true and nil error", func() {
+				certPem, keyPem, err := generateMockCertPEMWithExpiration("test", "test", time.Now().AddDate(200, 0, 0))
+				Expect(err).To(BeNil())
+
+				bootstrapClientConfig := &restclient.Config{
+					Host: "https://your-k8s-cluster.com",
+					TLSClientConfig: restclient.TLSClientConfig{
+						CertData: certPem,
+						KeyData:  keyPem,
+					},
+				}
+				err = writeKubeconfigFromBootstrapping(bootstrapClientConfig, validKubeConfigPath, false)
+				Expect(err).To(BeNil())
+
+				isValid, err := isKubeConfigStillValid(validKubeConfigPath, tlsBootstrapClient.logger)
+				Expect(isValid).To(Equal(true))
+				Expect(err).To(BeNil())
+			})
+		})
+
+		When("kubeconfg is empty", func() {
+			It("should return false and have error", func() {
+				_, err := os.Create(validKubeConfigPath) // overwrite the kubeconfig file with an empty file
```

Comment:
probably don't need to do this since we do the same in the BeforeEach?

Reply:
might be worth just doing the file creation in each test instead of in the BeforeEach so you can properly cleanup the file resource itself - e.g.:

f, err := os.Create(...)
Expect(err).To(BeNil())
defer f.Close()

usually not good practice to just ignore the returned *File handle

Reply:
actually, maybe https://pkg.go.dev/os#CreateTemp would be better so we don't have to worry about potential weirdness with the same file being used by each test case?

Reply:
from the example I linked above:

	f, err := os.CreateTemp("", "example")
	if err != nil {
		log.Fatal(err)
	}
	defer os.Remove(f.Name()) // clean up